build/TestNuGetsTasks/TestMonoGameTemplateTaskBase.cs

```diff
@@ -0,0 +1,597 @@
+using System.Text.RegularExpressions;
+using System.IO;
+using IOPath = System.IO.Path;
+
+namespace BuildScripts;
+
+public abstract class TestMonoGameTemplateTaskBase : FrostingTask<BuildContext>
+{
+    // Pre-compiled regex patterns for better performance
+    private static readonly Regex PackageReferenceRegex = new(@"<PackageReference\s+Include=""(MonoGame\.[^""]*)""\s+Version=""([^""]*)""\s*/?(?:\s*/>|>)", RegexOptions.Compiled);
```

Comment:
Use the following regex instead:
```
<PackageReference\s+?Include=""(?<packageName>MonoGame\.[^""]+?)""\s+?Version=""(?<version>[^""]+?)"".+?>
```

Then you can access `packageName` and `version` named groups instead of accessing using an index with `[0]` etc.